### PR TITLE
configured DB auth, endpoints access; added Role entity

### PR DESCRIPTION
**Pull request by TimoxaV**

### CODE REVIEW

File: src/main/java/com/cinema/service/DataInjectService.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        adminUser.setRoles(Set.of(roleAdmin, roleUser));
+        userService.add(adminUser);
+        shoppingCartService.registerNewShoppingCart(adminUser);
+        User user = new User();
+        user.setEmail("user@i.ua");
+        user.setPassword("111");
+        user.setRoles(Set.of(roleUser));
+        userService.add(user);
+        shoppingCartService.registerNewShoppingCart(user);
+    }
```
Comment (fyarkiy) [inline src/main/java/com/cinema/service/DataInjectService.java:43]: Don't know why would you want to add dummy user in your DB (from business logic). But if you do - I would consider to use (to call) AuthService

**Review by fyarkiy — APPROVED** [review]

File: src/main/java/com/cinema/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    private final UserService userService;
+
+    @Autowired
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> userOptional = userService.findByEmail(email);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/security/CustomUserDetailsService.java]: just use here orElseThrow with UsernameNotFoundException

Comment (TimoxaV) [reply src/main/java/com/cinema/security/CustomUserDetailsService.java]: done

File: src/main/java/com/cinema/security/CustomUserDetailsService.java [hunk 3]
```diff
@@ -20,19 +19,15 @@ public CustomUserDetailsService(UserService userService) {
 
     @Override
     public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
-        Optional<User> userOptional = userService.findByEmail(email);
+        User user = userService.findByEmail(email).orElseThrow(
+                () -> new UsernameNotFoundException("User not found."));
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/security/CustomUserDetailsService.java]: ideally add email value to exception

**Review by RomanDubovskyi — APPROVED** [review]

File: src/main/java/com/cinema/security/CustomUserDetailsService.java (hunk 3 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/security/CustomUserDetailsService.java]: ok, thanks

**Pull request merged**